spec/system/consumer/split_checkout_spec.rb

```diff
@@ -708,6 +708,93 @@
         end
       end
 
+      describe "vouchers" do
+        context "with no voucher available" do
+          before do
+            visit checkout_step_path(:payment)
+          end
+
+          it "doesn't show voucher input" do
+            expect(page).not_to have_content "Apply voucher"
+          end
+        end
+
+        context "with voucher available" do
+          let!(:voucher) { Voucher.create(code: 'some_code', enterprise: distributor) }
+
+          before do
+            visit checkout_step_path(:payment)
+          end
+
+          it "shows voucher input" do
+            expect(page).to have_content "Apply voucher"
+          end
+
+          describe "adding voucher to the order" do
+            shared_examples "adding voucher to the order" do
+              before do
+                fill_in "Enter voucher code", with: voucher.code
+                click_button("Apply")
+              end
+
+              it "adds a voucher to the order" do
+                expect(page).to have_content("$10.00 Voucher")
+                expect(order.reload.voucher_adjustments.length).to eq(1)
+              end
+            end
+
+            it_behaves_like "adding voucher to the order"
+
+            context "when voucher covers more then the order total" do
+              before do
+                order.total = 6
+                order.save!
+              end
+
+              it_behaves_like "adding voucher to the order"
+
+              it "shows a warning message" do
+                fill_in "Enter voucher code", with: voucher.code
+                click_button("Apply")
```

Comment:
I personally find shared examples difficult to follow. I often find additional methods helpful. 
```rb
def apply_voucher(voucher)
  fill_in...
  click...
end

def expect_voucher_applied
  ...
end
```
```

Reply:
I am not a big fan either it's not always clear where they are defined, but I do like the readability of `it_behaves_like` :man_shrugging: